src/Listener/ViewListener.php

```diff
@@ -318,18 +318,23 @@ protected function _getControllerActions()
 
         $actions = $this->_getAllowedActions();
         foreach ($actions as $actionName) {
-            $action = $this->_action($actionName);
             $method = 'GET';
-            $class = get_class($action);
-            $class = substr($class, strrpos($class, '\\') + 1);
-            $scope = $action->scope();
-
-            if ($class === 'DeleteAction') {
-                $method = 'DELETE';
-            }
+            try {
```

Comment:
you can do `$this-> _crud()-> isActionMapped()` instead - faster due to not having an exception thrown

https://github.com/FriendsOfCake/crud/blob/master/src/Controller/Component/CrudComponent.php#L378